### PR TITLE
[WIP] Persist User data

### PR DESCRIPTION
**Pull request by m-pereira**
### Overview

* configure redux
* connect SignIn component, to send userData to Store
* connect Dashboard component, and get userData from store

### Screenshots

deepin-screen-recorder_Select area_20210514002028

closes: #6

### CODE REVIEW

File: src/pages/login/login.component.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import CircularProgress from '@material-ui/core/CircularProgress';
 import signIn from '../../service/auth';
 import { makeStyles } from '@material-ui/core/styles';
 import { useHistory } from 'react-router-dom';
+import { connect } from 'react-redux';
 
 const INITIAL_STATE = {
```
Comment (belgamo) [inline src/pages/login/login.component.js:17]: Why this is here?

File: src/pages/login/login.component.js [hunk 2]
```diff
@@ -148,4 +154,4 @@ const SignIn = () => {
   );
 };
 
-export default SignIn;
+export default connect()(SignIn);
```
Comment (belgamo) [inline src/pages/login/login.component.js:157]: Instead connect store to the components, what do you think about `useSelector` hook?

File: src/pages/dashboard/dashboard.component.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        ))}
+      </ul>
+      <button type='button' onClick={handleSignOut}>
+        Logout
+      </button>
+    </>
+  );
+};
+
+const mapStateToProps = (state) => ({ currentUser: state.currentUser });
```
Comment (belgamo) [inline src/pages/dashboard/dashboard.component.js:40]: Same here, use `useSelector` hook instead.

File: src/pages/dashboard/dashboard.component.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+import { useHistory } from 'react-router-dom';
+
+const Dashboard = () => {
+  const currentUser = useSelector((state) => state.currentUser);
+  const dispatch = useDispatch();
+  const history = useHistory();
+
+  const handleSignOut = () => {
+    dispatch({
+      type: '@currentUser/SIGN_OUT',
```
Comment (belgamo) [inline src/pages/dashboard/dashboard.component.js:12]: What do you think about create a file exporting the constants instead working with strings directly?

File: src/store/modules/currentUser/reducer.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const currentUser = (state = {}, action) => {
```
Comment (belgamo) [inline src/store/modules/currentUser/reducer.js:1]: Set the initial state.

File: src/store/modules/rootReducer.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import { combineReducers } from 'redux';
+import currentUser from './currentUser/reducer';
```
Comment (belgamo) [inline src/store/modules/rootReducer.js:2]: Change the name to `userReducer`. It makes more sense.

File: src/store/modules/currentUser/reducer.js (hunk 5 above)
Comment (m-pereira) [reply src/store/modules/currentUser/reducer.js]: thanks

File: src/store/modules/rootReducer.js (hunk 6 above)
Comment (m-pereira) [reply src/store/modules/rootReducer.js]: it's done

File: src/store/modules/currentUser/reducer.js (hunk 5 above)
Comment (m-pereira) [reply src/store/modules/currentUser/reducer.js]: Noice!

**Review by belgamo — CHANGES_REQUESTED** [review]

**Review by belgamo — APPROVED** [review]

**Pull request merged**